src/Branching/perform-migration-from-svn-to-git.md

```diff
@@ -0,0 +1,183 @@
+
+> [!TIP]
+> *bare* Git repositories are structured differently and given the fact that it doesn't have a working directory prevent direct commit to the repository.
+>![bare git repo](_img/perform-migration-from-svn-to-git/bare-git-repo.png)
+
+### Retrieve a list of all Subversion authors
+
+Subversion just uses the username for each commit. Git stores more data for each author but at a minimal, each commit will need a name and an email. By default, The git-svn tool will list the SVN username in the author and email fields. However, you can create a mapping file for SVN users along with their corresponding Git names and emails.
+
+*Subversion users*
+![Subversion users](_img/perform-migration-from-svn-to-git/svn-log.png)
+
+*Git users*
+![git users](_img/perform-migration-from-svn-to-git/git-log.png)
+
+To extract a list of all SVN users, from the root of your local Subversion checkout, run this command:
+
+```
+svn log -q | awk -F '|' '/^r/ {sub("^ ", "", $2); sub(" $", "", $2); print $2" = "$2" <"$2">"}' | sort -u > authors-transform.txt
+```
+This command will retrieve all the log messages, extract the usernames, eliminate any duplicate usernames, sort the usernames and place them into a “authors-transform.txt” file. You can then edit each line in the file to create a mapping of SVN users to a well-formatted Git user. For example, you can map `almrangers = almrangers <almrangers> ` to `almrangers =  almrangers <almrangers@example.com>` .
+
+### Clone the Subversion repository using git-svn
+
+The following command will do the standard git-svn transformation using the authors-transform.txt file created in the previous step. It will place the Git repository in the "c:\mytempdir" folder in your local machine.
+```
+git svn clone ["SVN repo URL"] --prefix=svn/ --no-metadata --authors-file "authors-transform.txt" --stdlayout c:\mytempdir
+```
+
+> [!NOTE]
+>
+> The –prefix=svn/ is necessary because otherwise the tools can’t tell apart SVN revisions from imported ones. Setting a prefix (with a trailing slash) is strongly encouraged in any case, as your SVN-tracking refs will then be located at "refs/remotes/$prefix/", which is compatible with Git’s own remote-tracking ref layout (refs/remotes/$remote/).
+>
+> Setting a prefix is also useful if you wish to track multiple projects that share a common repository. By default, the prefix is set to origin/.
+
+If you are using the standard trunk, branches, tags layout you’ll just put –stdlayout. However if you have something different you may have to pass the –trunk, –branches, and –tags in to identify what is what. For example if your repository structure was trunk/companydir and you branched that instead of trunk, you would probably want ‘–trunk=trunk/companydir –branches=branches‘.
```

Comment:
Suggest code blocks around all command line options (eg `-stdlayout`) and folder names (eg `trunk/companydir`).

Reply:
done